docs/rfcs/0005-coupons.md

```diff
@@ -0,0 +1,102 @@
+## Title for RFC
+
+coupons
+
+### Abstract
+
+This describes the various kinds of coupons and how they will be implemented.
+
+Note: For this RFC the "next" course run is the course run which the user would be enrolled
+in when they enroll in a course, whether it is currently running or will run soon.
+
+##### Discount
+
+Coupons may have a percent discount. There is no fixed discount use case at the moment.
+
+##### Basket
+
+There can only be one coupon per basket, and a user may add at most only one course run,
+course, or program to a basket. This simplifies the UX and means we don't need to consider
+how multiple coupons can conflict.
+
+Users ultimately purchase course runs. In the UI when a user buys a course it will add the
+next course run to the basket. When a user buys a program it will add the next course run 
+for each course to the basket.
+
+##### Coupon target
+
+Coupons may apply to programs, courses, or course runs. The logic for what targets a coupon
+is redeemable for is different from MicroMasters.
+
+A program coupon is redeemable only if all courses for the program are in the basket. If a course
+was missing from the basket, maybe because the program is nearing end of life and one of
+the courses is not enrollable anymore, then the program coupon would not be redeemable at all.
+  
+A course coupon would be redeemable only for the next course run for that course. It may have 
+multiple courses where it can be redeemed.
```

Comment:
@noisecapella not sure I understand this

Reply:
A coupon can have multiple courses, programs or course runs where it could be redeemed. For example, you could have a 25% off coupon which could work for all of the courses of a program. This is different from a program coupon because that would require all courses be purchased in a bundle, but a course coupon could be set up to let the user apply the discount to any particular course.